src/ui/routes/admin.js

```diff
@@ -178,6 +180,77 @@ function mustBeAdminOrApproverMiddleware(req, res, next) {
     return next();
 }
 
+function getBaseUri(req) {
+    let baseUrl = "http://localhost:8001";
+    if (req.app.portalGlobals.network.kongAdminUrl) {
+        baseUrl = req.app.portalGlobals.network.kongAdminUrl;
+    }
+    return baseUrl;
+}
+
+function getAdmin(req, res, uri, callback) {
+    let baseUrl = getBaseUri(req);
+    request.get(
+        {
+            url: baseUrl + uri
+        },
+        callback);
+}
+
+function getSubscriptions(req, res, next) {
+    const filterFields = ['application', 'application_name', 'plan', 'api', 'owner', 'user'];
+    const subsUri = utils.makePagingUri(req, '/subscriptions?embed=1&', filterFields);
+    utils.getFromAsync(req, res, subsUri, 200, function (err, subsResponse) {
+        if (err)
+            return next(err);
+        if (!utils.acceptJson(req)) {
+            res.render('admin_subscriptions', {
+                authUser: req.user,
+                glob: req.app.portalGlobals,
+                title: 'All Subscriptions',
+            });
+            return;
+        }
+        if (utils.acceptJson(req)) {
+            res.json({
+                title: 'All Subscriptions',
+                subscriptions: subsResponse
+            });
+        }
+    });
+}
+
+function getFilteredApiKeyId(req, res, next, callback) {
+    const apikey = req.query.apikey.trim();
+    const filterFields = ['application_name', 'plan', 'api', 'owner', 'user', 'id'];
+    getAdmin(req, res, `/key-auths/${apikey}/consumer`, (err, Apikey) => {
+        if (err) {
+            return next(err);
+        }
+        let body = utils.getJson(Apikey.body);
+        if (body && body.message === "Not found") {
+            res.json({
+                title: 'All Subscriptions',
+                subscriptions: null
+            });
+        } else {
+            debug(`i am going inside else `)
```

Comment:
either remove this debug statement or out some meaningful message

Reply:
okay